### DISCUSSION TITLE
".." links are confusing when using index routes

### DISCUSSION THREAD

**Discussion by srmagura** [remix-run/react-router#8998] [Q&A] [answered]
From the [`Link` component docs](https://reactrouter.com/docs/en/v6/components/link):

> It may contain `..` to link to routes further up the hierarchy. In these cases, `..` works exactly like the command-line cd function; each `..` removes one segment of the parent path.
>
> Note:
> `<Link to>` with a .. behaves differently from a normal `<a href>` when the current URL ends with /. `<Link to>` ignores the trailing slash, and removes one URL segment for each `..`.

This suggests that, to go from `/settings` to `/`, I should render a `<Link to=".." />`. But the following example shows that `<Link to="../.." />` is actually required if `/settings` renders an index route.

Is this working as intended? Do you agree that the real behavior seems to differ from what you would intuitively expect and what is stated in the documentation?

[CodeSandbox](https://codesandbox.io/s/react-router-dotdot-links-2jyt68)

```tsx
import { BrowserRouter, Routes, Route, Link } from "react-router-dom";

export default function App() {
  return (
    <BrowserRouter>
      <Routes>
        <Route path="/">
          <Route
            index
            element={
              <div>
                <p>Home</p>
                <Link to="settings">Settings</Link>
              </div>
            }
          />
          <Route path="settings">
            <Route
              index
              element={
                <div>
                  <p>Settings</p>
                  <p>
                    <Link to="account">Account Settings</Link>
                  </p>
                  <p>
                    {/* Why does this need ../.. instead of just .. to work? */}
                    <Link to="../..">Go back</Link>
                  </p>
                </div>
              }
            />
            <Route
              path="account"
              element={
                <div>
                  <p>Account Settings</p>
                  <Link to="..">Go back</Link>
                </div>
              }
            />
          </Route>
        </Route>
      </Routes>
    </BrowserRouter>
  );
}
```

**Comment by prajotKeywordio** [comment]
I am also having this issue. Did you manage to find a solution to this ?

**Reply by srmagura** [reply]
Not really. I just use `<Link to="../..">` when I have to.

**Comment by brophdawg11** [comment, marked as answer]
This should be resolved in `6.4.0` - see #8350 👍
